src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,902 @@
+			var userSize = GetUserSizes();
+			var strokeThickness = StrokeThickness;
+			var halfStrokeThickness = GetHalfStrokeThickness();
+			var pathBounds = path.PathBoundingBox; // The BoundingBox does also contains bezier anchors even if out of geometry
+			var pathSize = (Size)pathBounds.Size;
+
+			if (nfloat.IsInfinity(pathBounds.Right) || nfloat.IsInfinity(pathBounds.Bottom))
+			{
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Ignoring path with invalid bounds {pathBounds}");
+				}
+
+				return default;
+			}
+
+			// Compute the final size of the Shape and the render properties
+			Size size;
+			(double x, double y) renderOrigin, renderScale, renderOverflow;
+			switch (stretch)
+			{
+				default:
+				case Stretch.None:
+					var pathNaturalSize = new Size(
+						pathBounds.X == 0 ? pathBounds.Width + strokeThickness : pathBounds.Right + halfStrokeThickness,
+						pathBounds.Y == 0 ? pathBounds.Height + strokeThickness : pathBounds.Bottom + halfStrokeThickness);
+					var (userMinSize, userMaxSize) = GetMinMax(userSize);
+
+					size = pathNaturalSize.AtMost(userMaxSize).AtLeast(userMinSize); // The size defined on the Shape has priority over the size of the geometry itself!
+					renderScale = (1, 1);
+					renderOrigin = (0, 0);
+					renderOverflow = (size.Width - finalSize.Width, size.Height - finalSize.Height); // We do not add halfStrokeThickness: The stroke is allowed to flow out of container for None
+					break;
+
+				case Stretch.Fill:
+					size = ComputeSizeLowerThanBounds(userSize, finalSize);
+					renderScale = ComputeScaleFactors(pathSize, ref size, strokeThickness);
+					renderOrigin = (halfStrokeThickness - pathBounds.X * renderScale.x, halfStrokeThickness - pathBounds.Y * renderScale.y);
+					renderOverflow = (size.Width - finalSize.Width, size.Height - finalSize.Height);
+					break;
+
+				case Stretch.Uniform:
+#if !IS_DESIRED_SMALLER_THAN_CONSTRAINTS_ALLOWED
+					var boundsAdjustements = AdjustRenderingBounds(userSize, ref finalSize, horizontal, vertical);
+#endif
+					var defaultSize = size = ComputeSizeLowerThanBounds(userSize, finalSize);
+
+					// This set of rules are a complete non sense as we should normally just use userSize.min.width and userSize.min.height,
+					// but they are actually reproducing a bug of WinUI where the MinWidth and MinHeight are sometimes
+					// constrained by the layout slot ...
+					// Note: This is only a replication of what we observed in the UI tests, and might have some flaw in the logic.
+					//		 Especially, we expect that the max vs. min applied on Width vs. Height is probably drove by the aspect ratio.
+					var minSizeForScale = default(Size);
+					if (userSize.min.hasWidth && userSize.min.hasHeight)
+					{
+						var min = Math.Min(userSize.min.width, userSize.min.height);
+						minSizeForScale = new Size(0, min);
+					}
+					else if (userSize.min.hasWidth && this is Path)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5633443)